File: The Trip/trip.java
import java.util.*;

public class Main {

    public static double minMoneyExchange(double[] expenses) {
        double totalExpenses = 0;
        for (double expense : expenses) {
            totalExpenses += expense;
        }
        System.out.println(totalExpenses);

        double equalShare = totalExpenses / expenses.length;

        double moneyExchange = 0;
        for (double expense : expenses) {
            moneyExchange += Math.abs(expense - equalShare);
        }

        return moneyExchange;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter no. of friends: ");
        int n = scanner.nextInt();

        double[] expenses = new double[n];
        System.out.print("Enter expenses:");
        for (int i = 0; i < n; ++i) {
            expenses[i] = scanner.nextDouble();
        }

        double minMoney = minMoneyExchange(expenses);
        System.out.println(Math.round(minMoney / 2 * 100) / 100);

        scanner.close();
    }
}

File: The Trip/trip.py
def min_money_exchange(expenses):
    total_expenses = sum(expenses)
    print(round(total_expenses, 2))

    equal_share = total_expenses / len(expenses)

    money_exchange = sum(abs(expense - equal_share) for expense in expenses)

    return money_exchange

expenses = list(map(float, input("Enter expenses: ").split()))
min_money = min_money_exchange(expenses)
print(round(min_money/2, 2))